d2layouts/d2grid/grid_diagram.go

```diff
@@ -66,6 +75,18 @@ func newGridDiagram(root *d2graph.Object) *gridDiagram {
 		gd.rowDirected = true
 	}
 
+	// grid gap sets both, but can be overridden
+	if root.Attributes.GridGap != nil {
+		gd.gapRows, _ = strconv.ParseFloat(root.Attributes.GridGap.Value, 64)
```

Comment:
why parse float? parse int. strconv.Atoi, like u did when validating

Reply:
why is the data type for gapRows float64? just to make it easier to work with width/height? naw, use int and convert when using

Reply:
i understand, but conforming to another data type just to make the operations easier is a pattern i disagree with.